crates/runtime/src/component/http.rs

```diff
@@ -132,90 +133,93 @@ where
             wasmtime_wasi_http::bindings::http::types::ErrorCode,
         >,
     > {
-        let span = cx.deref().clone();
-        async move {
-            let scheme = request.uri().scheme().context("scheme missing")?;
-            let scheme =
-                wrpc_interface_http::bindings::wrpc::http::types::Scheme::from(scheme).into();
+        // Set the parent of the current context to the span passed in
+        Span::current().set_parent(cx.deref().context());
+        let scheme = request.uri().scheme().context("scheme missing")?;
+        let scheme = wrpc_interface_http::bindings::wrpc::http::types::Scheme::from(scheme).into();
 
-            let (tx, rx) = oneshot::channel();
-            let mut store = new_store(&self.engine, self.handler.clone(), self.max_execution_time);
-            let pre = incoming_http_bindings::IncomingHttpPre::new(self.pre.clone())
-                .context("failed to pre-instantiate `wasi:http/incoming-handler`")?;
-            trace!("instantiating `wasi:http/incoming-handler`");
-            let bindings = pre
-                .instantiate_async(&mut store)
-                .await
-                .context("failed to instantiate `wasi:http/incoming-handler`")?;
-            let data = store.data_mut();
+        let (tx, rx) = oneshot::channel();
+        let mut store = new_store(&self.engine, self.handler.clone(), self.max_execution_time);
+        let pre = incoming_http_bindings::IncomingHttpPre::new(self.pre.clone())
+            .context("failed to pre-instantiate `wasi:http/incoming-handler`")?;
+        trace!("instantiating `wasi:http/incoming-handler`");
+        let bindings = pre
+            .instantiate_async(&mut store)
+            .instrument(debug_span!("instantiate_async"))
+            .await
+            .context("failed to instantiate `wasi:http/incoming-handler`")?;
+        let data = store.data_mut();
 
-            // The below is adapted from `WasiHttpView::new_incoming_request`, which is unusable for
-            // us, since it requires a `hyper::Error`
+        // The below is adapted from `WasiHttpView::new_incoming_request`, which is unusable for
+        // us, since it requires a `hyper::Error`
 
-            tracing::error!("HELLO");
+        let (parts, body) = request.into_parts();
+        let body = HostIncomingBody::new(
+            body,
+            // TODO: this needs to be plumbed through
+            std::time::Duration::from_millis(600 * 1000),
```

Comment:
Should this be `std::time::Duration::from_mins(10)` instead?

Reply:
Yes, but this is copy-pasted from upstream, so let's keep it as is